File: bro-agent-api/src/main/python/bro_agent/agent/BackupExecutionState.py
import bro_agent.agent.states

from bro_agent.agent.ActionInformation import ActionInformation
from bro_agent.agent.BackupExecutionActions import BackupExecutionActions
from bro_agent.generated.Action_pb2 import Action


class BackupExecutionState(bro_agent.agent.AgentState.AgentState):
    def __init__(self, agent, preparation):
        bro_agent.agent.AgentState.AgentState.__init__(self, agent)
        self.action_information = ActionInformation.from_preparation(preparation)

    def process_message(self, message):
        return bro_agent.agent.WaitingForActionState.WaitingForActionState(self.agent)

    def trigger(self):
        self.agent.execute_backup(BackupExecutionActions(self.agent, self.action_information))
        self.agent.finish_action()

    def finish_action(self):
        return bro_agent.agent.FinishedActionState.FinishedActionState(self.agent, self.action_information.backup_name, Action.BACKUP)

    def cancel_action(self):
        return bro_agent.agent.WaitingForActionState.WaitingForActionState(self.agent)

File: bro-agent-api/src/main/java/com/ericsson/adp/mgmt/bro/api/agent/BackupExecutionState.java
/**------------------------------------------------------------------------------
 *******************************************************************************
 * COPYRIGHT Ericsson 2019
 *
 * The copyright to the computer program(s) herein is the property of
 * Ericsson Inc. The programs may be used and/or copied only with written
 * permission from Ericsson Inc. or in accordance with the terms and
 * conditions stipulated in the agreement/contract under which the
 * program(s) have been supplied.
 *******************************************************************************
 *------------------------------------------------------------------------------*/

package com.ericsson.adp.mgmt.bro.api.agent;

import static com.ericsson.adp.mgmt.action.Action.BACKUP;
import static com.ericsson.adp.mgmt.control.OrchestratorMessageType.POST_ACTIONS;

import com.ericsson.adp.mgmt.control.OrchestratorControl;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

/**
 * Represents an Agent performing a backup.
 */
public class BackupExecutionState extends BackupState {

    private static final Logger log = LogManager.getLogger(BackupExecutionState.class);

    /**
     * @param agent The agent performing the backup
     * @param actionInformation information required for backup
     */
    public BackupExecutionState(final Agent agent, final ActionInformation actionInformation) {
        super(agent, actionInformation);
    }

    @Override
    public AgentState processMessage(final OrchestratorControl message) {
        if (isPostActionsMessage(message)) {
            log.info("Received backup post action message for backup: {}", actionInformation.getBackupName());
            return new PostBackupState(agent, actionInformation);
        } else if (isCancelMessage(message)) {
            log.info("Received cancel backup message for backup: {}", actionInformation.getBackupName());
            return cancelAction();
        }
        agent.sendStageCompleteMessage(false, "Unsuccessful backup stage.", BACKUP);
        return new WaitingForActionState(agent);
    }

    private boolean isPostActionsMessage(final OrchestratorControl message) {
        return message.getOrchestratorMessageType().equals(POST_ACTIONS) && message.hasPostActions();
    }

    @Override
    public void trigger() {
        agent.executeBackup(new BackupExecutionActions(agent, actionInformation));
    }

    @Override
    public AgentState finishAction() {
        this.agent.sendStageCompleteMessage(false, "Finish action called during backup execution", BACKUP);
        return new FinishedActionState(this.agent, this.actionInformation.getBackupName(), BACKUP);
    }
}
